IdentityCore/src/workplacejoin/MSIDPkeyAuthHelper.m

```diff
@@ -32,20 +32,35 @@
 #import "MSIDWorkplaceJoinChallenge.h"
 #import "MSIDAADTokenRequestServerTelemetry.h"
 #import "MSIDADFSAuthority.h"
+#import "MSIDExternalSSOContext.h"
 
 @implementation MSIDPkeyAuthHelper
 
 + (nullable NSString *)createDeviceAuthResponse:(nonnull NSURL *)authorizationServer
                                   challengeData:(nullable NSDictionary *)challengeData
+                             externalSSOContext:(MSIDExternalSSOContext *)externalSSOContext
                                         context:(nullable id<MSIDRequestContext>)context
 {
     NSString *authToken = @"";
     NSString *challengeContext = challengeData ? [challengeData valueForKey:@"Context"] : @"";
     NSString *challengeVersion = challengeData ? [challengeData valueForKey:@"Version"] : @"";
     NSString *challengeTenantId = challengeData ? [challengeData valueForKey:@"TenantId"] : @"";
     
-    MSIDWPJKeyPairWithCert *info = [MSIDWorkPlaceJoinUtil getWPJKeysWithTenantId:challengeTenantId context:context];
+    MSIDWPJKeyPairWithCert *info = nil;
     
+    if (externalSSOContext)
+    {
+        // First try reading WPJ from the SSO context if present
+        info = [MSIDWorkPlaceJoinUtil wpjKeyPairWithSSOContext:externalSSOContext tenantId:challengeTenantId context:context];
+    }
+    
+    if (!info)
+    {
+        // If no SSO context is present, or mismatches requested tenant, try using old way
```

Comment:
if ssoContext mismatches requested tenant , do we want to try the old way ? Are we supporting both registrations being present at the same time on a given device  (platform sso and keychain)?

Reply:
Maybe won't be needed in the future, I'm a bit uncomfortable relying on it solely for now, so would like to keep both lookups